sale_prequotation/sale_prequotation.py

```diff
@@ -869,6 +874,25 @@ def action_sale(self, cr, uid, ids, format_quote, context=None):
         result['domain'] = "[('id','in', [" + ','.join(map(str, quote_ids)) + "])]"
         return result
 
+    def action_new_pq_version(self, cr, uid, ids, context=None):
+        for prequote in self.browse(cr, uid, ids, context=context):
+            pq_new = self.copy(cr, uid, prequote.id, context=context)
+            self.write(cr, uid, self.search(cr, uid, [('id', '=', pq_new)]), {
+                'name': prequote.name + '-' + NUMBER_TO_CHAR[prequote.pq_version],
+                'pq_id': prequote.id,
+                'date_pq': prequote.date_pq,
```

Comment:
Why need so many update?

Reply:
cuz in def copy(

values is set to this:
```default.update({
            'date_pq': time.strftime('%Y-%m-%d'),
            'cal_date': time.strftime('%Y-%m-%d'),
            'state': 'draft',
            'user_id': uid,
            'sale_ids': False,
            'mat_sale_id': None,
            'labour_sale_id': None,
            'doc_version': 0,
            'mat_sale_no': False,
            'labour_sale_no': False,
            'pq_history_ids': False,
            'name': self.pool.get('ir.sequence').get(cr, uid, 'sale.prequotation') or '/',
        })```